### DISCUSSION TITLE
BadRequest for different actions

### DISCUSSION THREAD

**Discussion by PRCV1** [pocketbase/pocketbase#1093] [Q&A] [answered]
Hello,

while updating my C# SDK to support the new user/admin system, i get 400 Bad Request for some actions:

/api/admins/confirm-password-reset
/api/collections/users/confirm-password-reset
/api/collections/users/confirm-email-change

thats the json response from confirm-password-reset:
`{"code":400,"message":"Failed to set new password.","data":{"token":{"code":"validation_invalid_token","message":"Invalid or expired token."}}}`

and the jwt-token:
`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NzAzNjIzODYsImlkIjoib25nMXg1djB0OXRxN21nIiwidHlwZSI6ImFkbWluIn0.CqKF1-eIN50COd3dPpIrrifIdpVZXsZzZfcm2U9SXYo`

according to jwt.io the token is valid and not expired.

please tell me if you need more information :)

**Comment by ganigeorgiev** [comment, marked as answer]
From the token payload it doesn't look like you are using a password-reset or confirm-email change token (aka. the token that is available in the url after the user click on the email).

PocketBase uses different secrets for each token types (auth, verification, password reset, email change, etc.).

**Reply by PRCV1** [reply]
ouh, yes, you are right. i somehow screwed my tests up for these methods, instead of sending the token from the url i was sending the jwt from the authenticate-method. 

well, maybe i shouldnt work on this sdk in the evening after 8hrs of work in the office. thank you.

sidequestion:
should i create a feature request to show the jwt-token in the logs sidepanel? does this make sense? at least for me while developing the C# sdk.

**Reply by ganigeorgiev** [reply]
> should i create a feature request to show the jwt-token in the logs sidepanel? does this make sense? at least for me while developing the C# sdk.

This may not be desirable by everyone and can open the application to some vulnerabilities if a malicious actor somehow get their hands on the logs.db (eg. from an external backup).
